src/number_theory/pell.lean

```diff
@@ -38,6 +40,23 @@ Pell's equation
 * Connect solutions to the continued fraction expansion of `√d`.
 -/
 
+namespace zsqrtd
+
+-- Note: We put this here to avoid having to import `algebra.star.unitary`
+--       into `number_theory.zsqrtd.basic` just for this one result.
+
+/-- An element of `ℤ√d` has norm one (i.e., `a.re^2 - d*a.im^2 = 1`) if and only if
+it is contained in the submonoid of unitary elements. -/
+lemma norm_eq_one_iff_mem_unitary {d : ℤ} {a : zsqrtd d} :
+  a.re ^ 2 - d * a.im ^ 2 = 1 ↔ a ∈ unitary (zsqrtd d) :=
```

Comment:
```suggestion
  a.norm = 1 ↔ a ∈ unitary (zsqrtd d) :=
```

Reply:
This would then require an additional rewriting step in `rel` and in `mk`.
If you think the proposed version should be there, I can add it and rename the other one to `norm_eq_one_iff_mem_unitary'`.

Reply:
I guess I'll just do that.